cmd/terramate/cli/gitlab/mr.go

```diff
@@ -0,0 +1,172 @@
+// Copyright 2024 Terramate GmbH
+// SPDX-License-Identifier: MPL-2.0
+
+package gitlab
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"net/http"
+	"net/url"
+	"os"
+
+	"github.com/terramate-io/terramate/errors"
+	"github.com/terramate-io/terramate/printer"
+)
+
+const (
+	// ErrNotFound indicates the resource does not exists.
+	ErrNotFound errors.Kind = "resource not found (HTTP Status: 404)"
+)
+
+type (
+	// Client is a Gitlab API HTTP client wrapper.
+	Client struct {
+		// BaseURL is the base URL used to construct the final URL of endpoints.
+		// If not set, it uses CI_API_V4_URL environment variable.
+		BaseURL string
+
+		// HTTPClient sets the HTTP client used and then allows for advanced
+		// connection reuse schemes. If not set, a new http.Client is used.
+		HTTPClient *http.Client
+
+		// Token is the Gitlab token (usually provided by the GITLAB_TOKEN environment variable.
+		Token string
+
+		ProjectID int
+		Group     string
+		Project   string
+	}
+
+	// MRs is a list of Gitlab Merge Requests.
+	MRs []MR
+
+	// MR is a Gitlab Merge Request.
+	MR struct {
+		ID             int      `json:"id"`
+		IID            int      `json:"iid"`
+		ProjectID      int      `json:"project_id"`
+		Title          string   `json:"title"`
+		Description    string   `json:"description,omitempty"`
+		State          string   `json:"state"`
+		CreatedAt      string   `json:"created_at,omitempty"`
+		UpdatedAt      string   `json:"updated_at,omitempty"`
+		TargetBranch   string   `json:"target_branch"`
+		SourceBranch   string   `json:"source_branch"`
+		Upvotes        int      `json:"upvotes"`
+		Downvotes      int      `json:"downvotes"`
+		Author         User     `json:"author"`
+		Labels         []string `json:"labels"`
+		Draft          bool     `json:"draft"`
+		WorkInProgress bool     `json:"work_in_progress"`
+		SHA            string   `json:"sha"`
+		MergeStatus    string   `json:"merge_status"`
+		WebURL         string   `json:"web_url"`
+
+		// fields below are available but not used:
+		// - assignee
+		// - milestone
+		// - merge_when_pipeline_succeeds
+		// - merge_commit_sha
+		// - squash_commit_sha
+		// - user_notes_count
+		// - discussion_locked
+		// - should_remove_source_branch
+		// - force_remove_source_branch
+		// - time_stats.time_estimate
+		// - time_statis.total_time_spent
+		// - time_stats.human_time_estimate
+		// - time_stats.human_total_time_spent
+	}
+
+	// User is a Gitlab user.
+	User struct {
+		ID        int    `json:"id"`
+		Username  string `json:"username"`
+		Name      string `json:"name"`
+		WebURL    string `json:"web_url"`
+		AvatarURL string `json:"avatar_url"`
+		State     string `json:"state"`
+	}
+)
+
+// LatestMRsForCommit returns a list with latest Merge Requests associated with the provided commit SHA.
+func (c *Client) LatestMRsForCommit(ctx context.Context, sha string, limit int) (mrs MRs, err error) {
```

Comment:
Could this be problematic since we don't handle pagination here? What if limit exceeds the number of entries support per page by Gitlab? Will it be a bad request? Or will it capped? First case could become a problem later since the function looks like it'll just work for any limit. If second case, could add a comment to the docs:
If limit exceeds the max number of entries per page (currently 20? 50?), it will be capped at that value.

Reply:
we are only interested in 1 commit. Maybe I can change this to `client.LastMRForCommit()` and then use `1` by default?